pennylane/transforms/split_non_commuting.py

```diff
@@ -11,42 +11,45 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 """
-Contains the tape transform that splits non-commuting terms
+Contains the tape transform that splits a tape into tapes measuring commuting observables.
 """
-from functools import reduce
 
-# pylint: disable=protected-access
-from typing import Callable, Sequence
+from functools import partial
+from typing import Callable, Dict, List, Sequence, Tuple, Union
 
 import pennylane as qml
+from pennylane.measurements import ExpectationMP, MeasurementProcess, Shots, StateMP
+from pennylane.ops import Hamiltonian, LinearCombination, Prod, SProd, Sum
 from pennylane.transforms import transform
-
-
-def null_postprocessing(results):
-    """A postprocesing function returned by a transform that only converts the batch of results
-    into a result for a single ``QuantumTape``.
-    """
-    return results[0]
+from pennylane.typing import Result, ResultBatch
 
 
 @transform
-def split_non_commuting(tape: qml.tape.QuantumTape) -> (Sequence[qml.tape.QuantumTape], Callable):
-    r"""
-    Splits a qnode measuring non-commuting observables into groups of commuting observables.
+def split_non_commuting(
+    tape: qml.tape.QuantumScript,
+    grouping: Union[bool, str] = True,
+) -> (Sequence[qml.tape.QuantumTape], Callable):
```

Comment:
```suggestion
) -> Tuple[Sequence[qml.tape.QuantumTape], Callable]:
```
Parentheses `()` aren't valid type-hinting syntax for tuples 😃